### ISSUE TITLE
[1.19.2  - 2.2-b04k]: Server hangs after teleporting to a location

### ISSUE THREAD

**Issue by SiverDX** [Momo-Studios/Cold-Sweat-Rework-Forge#153]
I was in `Specator` mode (not sure if that is relevant)

``` log
"Server thread" prio=4 Id=167 TIMED_WAITING on java.lang.String@8b20197
	at java.base@17.0.1/jdk.internal.misc.Unsafe.park(Native Method)
	-  waiting on java.lang.String@8b20197
	at java.base@17.0.1/java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.BlockableEventLoop.m_5667_(BlockableEventLoop.java:152)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:142)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerChunkCache.getChunkBlocking(ServerChunkCache.java:3262)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerChunkCache.m_7587_(ServerChunkCache.java:3158)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.chunk.ChunkSource.m_62227_(ChunkSource.java:14)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.util.world.WorldHelper.getChunk(WorldHelper.java:151)
	...
extended trace:
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.api.temperature.modifier.BlockTempModifier.lambda$calculate$0(BlockTempModifier.java:57)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.api.temperature.modifier.BlockTempModifier$$Lambda$40594/0x00000008055ee298.apply(Unknown Source)
	at java.base@17.0.1/java.util.HashMap.computeIfAbsent(Unknown Source)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.api.temperature.modifier.BlockTempModifier.calculate(BlockTempModifier.java:56)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.api.temperature.modifier.TempModifier.update(TempModifier.java:60)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.api.util.Temperature.apply(Temperature.java:82)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.api.util.Temperature.apply(Temperature.java:95)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.common.event.EntityTempHandler.lambda$initModifiersOnEntity$15(EntityTempHandler.java:266)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.common.event.EntityTempHandler$$Lambda$44693/0x0000000805b28000.run(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.BlockableEventLoop.execute(BlockableEventLoop.java:103)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.common.event.EntityTempHandler.initModifiersOnEntity(EntityTempHandler.java:247)
	at TRANSFORMER/cold_sweat@2.2-b04k/dev.momostudios.coldsweat.common.event.__EntityTempHandler_initModifiersOnEntity_EntityJoinLevelEvent.invoke(.dynamic)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus$$Lambda$4864/0x0000000801a50b00.invoke(Unknown Source)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:315)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:296)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157538_(PersistentEntitySectionManager.java:79)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157604_(PersistentEntitySectionManager.java:121)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.entity.PersistentEntitySectionManager$$Lambda$44401/0x0000000805aeaea8.accept(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.entity.EntityType$1.m_147060_(EntityType.java:529)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.entity.EntityType$1$$Lambda$39891/0x00000008054dd230.apply(Unknown Source)
	at java.base@17.0.1/java.util.Optional.map(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.entity.EntityType.m_20645_(EntityType.java:507)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.entity.EntityType$1.m_147056_(EntityType.java:528)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.entity.EntityType$1$$Lambda$39890/0x00000008054dd000.accept(Unknown Source)
	at java.base@17.0.1/java.util.ArrayList$ArrayListSpliterator.tryAdvance(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.entity.EntityType$1.tryAdvance(EntityType.java:527)
	at java.base@17.0.1/java.util.Spliterator.forEachRemaining(Unknown Source)
	at java.base@17.0.1/java.util.stream.ReferencePipeline$Head.forEach(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157559_(PersistentEntitySectionManager.java:120)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerLevel.m_143327_(ServerLevel.java:1517)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap.m_143064_(ChunkMap.java:666)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap.m_214898_(ChunkMap.java:683)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap$$Lambda$43710/0x0000000805a42468.m_196866_(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.world.level.chunk.LevelChunk.m_62952_(LevelChunk.java:429)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap.m_214854_(ChunkMap.java:691)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap$$Lambda$39751/0x00000008054b27a0.apply(Unknown Source)
	at MC-BOOTSTRAP/datafixerupper@5.0.28/com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162)
	at MC-BOOTSTRAP/datafixerupper@5.0.28/com.mojang.datafixers.util.Either$$Lambda$39752/0x00000008054a5620.apply(Unknown Source)
	at MC-BOOTSTRAP/datafixerupper@5.0.28/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at MC-BOOTSTRAP/datafixerupper@5.0.28/com.mojang.datafixers.util.Either.mapLeft(Either.java:162)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap.m_214851_(ChunkMap.java:675)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkMap$$Lambda$39748/0x00000008054b2120.apply(Unknown Source)
	at java.base@17.0.1/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base@17.0.1/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ChunkTaskPriorityQueueSorter$$Lambda$39738/0x00000008054b0000.run(Unknown Source)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:157)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_6367_(ServerChunkCache.java:535)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:131)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_7245_(ServerChunkCache.java:543)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.level.ServerChunkCache.m_8466_(ServerChunkCache.java:267)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:751)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:740)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:140)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:726)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:658)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244)
	at TRANSFORMER/minecraft@1.19.2/net.minecraft.server.MinecraftServer$$Lambda$39173/0x00000008053ed198.run(Unknown Source)
	at java.base@17.0.1/java.lang.Thread.run(Unknown Source)
```

I teleported there again and it keeps happening (this time it could load some parts though before the server started to hang)

**Biome**: minecraft:desert
![image](https://github.com/Momo-Studios/Cold-Sweat/assets/22003703/9b2d736e-730d-443e-aff1-59bbac98bdad)

**Comment by SiverDX** [comment]
Went back in and stopped spark profiler after a couple of seconds (before it started to hang again):

https://spark.lucko.me/CT9RR5WnYl

![image](https://github.com/Momo-Studios/Cold-Sweat/assets/22003703/6e56d91a-a4d7-461d-b7f8-df14bea8ee7e)

![image](https://github.com/Momo-Studios/Cold-Sweat/assets/22003703/2976c590-e76c-47f8-b292-6f020f9d40b2)

**Comment by SiverDX** [comment]
I emptied out the chameleon biome list and killed all entities, which stopped the lags

**Comment by MikulDev** [comment]
This is pretty much a duplicate of #143, #144, and #151. It has to do with Chameleons trying to load chunks in a way that causes the chunk loader to hang, and by extension the entire server. This will be fixed soon

**Comment by SiverDX** [comment]
you should maybe leave an issue open until the fix is there

**Comment by MikulDev** [comment]
I've already confirmed the issue and tested a working fix, but I will reopen it if there are more issues
